### PR TITLE
Revise Docker publish instructions to run commands in background

### PR DESCRIPTION
**Pull request by jathanism**
### Fixes: #n/a

- Docker build/push completes much more quickly by running commands in parallel
- Also remove the "test the images locally" as that is superseded by our actual integration test suite

### CODE REVIEW

File: nautobot/docs/development/release-checklist.md [hunk 1]
```diff
@@ -172,39 +172,19 @@ Build the images locally:
 ```no-highlight
 for ver in 3.6 3.7 3.8 3.9; do
   export INVOKE_NAUTOBOT_PYTHON_VER=$ver
-  invoke buildx --target final --tag networktocode/nautobot-py${INVOKE_NAUTOBOT_PYTHON_VER}:local
-  invoke buildx --target final-dev --tag networktocode/nautobot-dev-py${INVOKE_NAUTOBOT_PYTHON_VER}:local
+  invoke buildx --target final --tag networktocode/nautobot-py${INVOKE_NAUTOBOT_PYTHON_VER}:local &
+  invoke buildx --target final-dev --tag networktocode/nautobot-dev-py${INVOKE_NAUTOBOT_PYTHON_VER}:local &
```
Comment (glennmatthews) [inline nautobot/docs/development/release-checklist.md:176]: One downside of this change is that it won't be obvious if any of the builds fails (which I have had happen before due to Docker Desktop running out of disk space).

File: nautobot/docs/development/release-checklist.md [hunk 2]
```diff
@@ -172,39 +172,19 @@ Build the images locally:
...
-```
-
-Push the images to GitHub Container Registry and Docker Hub
 ```no-highlight
 docker login
 docker login ghcr.io
 for ver in 3.6 3.7 3.8 3.9; do
   export INVOKE_NAUTOBOT_PYTHON_VER=$ver
-  invoke docker-push main
+  invoke docker-push main &
```
Comment (glennmatthews) [inline nautobot/docs/development/release-checklist.md:187]: Same concern here.

**Review by glennmatthews — COMMENTED** [review]
Should this go to `develop` rather than `next`?

File: nautobot/docs/development/release-checklist.md (hunk 1 above)
Comment (jathanism) [reply nautobot/docs/development/release-checklist.md:176]: Ah I've never had that. Outside of using a more advanced solution this is really just a stop-gap until we wire this up to CI.

**Comment by bryanculver** [comment]
@jathanism doesn't look like this is relevant with the recent CI changes. Safe to close?

**Pull request closed without merging**